src/api/controller/api/publishedDataset.js

```diff
@@ -93,12 +93,18 @@ export const getRemovedPage = async ctx => {
 
 export const removeResource = async ctx => {
     const { uri, reason } = ctx.request.body;
-
-    ctx.body = await ctx.publishedDataset.hide(uri, reason);
+    const removedAt = ctx.request.body.removedAt ?? new Date();
+    ctx.hiddenResource.create({
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Shouldn't you _await_ this call?

```suggestion
    await ctx.hiddenResource.create({
```